cpu/src/stark.rs

```diff
@@ -76,6 +76,73 @@ where
             reduce::<AB>(&base, local.read_value_1()),
         );
 
+        // (TODO: we'd need to range check opcode_lo16, since it should be less thatn 16)
+        builder.assert_eq(
+            local.instruction.opcode,
+            local.instruction.opcode_lo16 * AB::Expr::one()
```

Comment:
Might be more clear to call this `opcode_lo4`, since `opcode_lo16` makes me think "the low 16 bits of `opcode`," whereas what this is intended to be is the low 4 bits of `opcode`. And then `opcode_hi16` could be renamed to `opcode_hi28`, meaning the high 28 bits of `opcode`, assuming that `opcode` is a 32-bit value.